modin/core/io/column_stores/parquet_dispatcher.py

```diff
@@ -28,23 +29,79 @@
 
 
 from modin.core.io.column_stores.column_store_dispatcher import ColumnStoreDispatcher
-from modin.utils import import_optional_dependency
+from modin.utils import import_optional_dependency, _inherit_docstrings
 
 
-class ParquetDispatcher(ColumnStoreDispatcher):
-    """Class handles utils for reading `.parquet` files."""
+class Dataset(ABC):
+    """
+    Abstract class that encapsulates Parquet engine-specific details.
 
-    @classmethod
-    def get_fs_and_files(cls, path, storage_options):
+    This class exposes a set of functions that are commonly used in the
+    `read_parquet` implementation.
+    """
+
+    @abstractproperty
+    def pandas_metadata(self):
+        """Return the pandas metadata of the dataset."""
+        pass
+
+    @abstractproperty
+    def columns(self):
+        """Return the list of columns in the dataset."""
+        pass
+
+    @abstractproperty
+    def files(self):
+        """Return the list of formatted file paths of the dataset."""
+        pass
+
+    @abstractproperty
+    def engine(self):
+        """Return string representing what engine is being used."""
+        pass
+
+    @abstractmethod
+    def row_groups_per_file(self):
+        """Return a list with the number of row groups per file."""
+        pass
+
+    @abstractmethod
+    def read_table_as_pandas(self, columns):
```

Comment:
Could we call this `to_pandas_dataframe`? `table` was confusing to me.

Reply:
Yeah that's probably a better name.